### PR TITLE
Add Node.js 14.3/14.8 and fix Node release statuses

### PR DESCRIPTION
**Pull request by saschanaz**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [x] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
   * https://nodejs.org/en/blog/release/v14.3.0/
   * https://nodejs.org/en/blog/release/v14.8.0/
- [x] Data: if you tested something, describe how you tested with details like browser and version
   * `node -p process.versions.v8`
- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any: #7209

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

File: browsers/nodejs.json [hunk 1]
```diff
@@ -255,13 +255,27 @@
...
           "release_date": "2020-07-21",
           "release_notes": "https://nodejs.org/en/blog/release/v14.6.0/",
           "status": "retired",
           "engine": "V8",
           "engine_version": "8.4"
         },
+        "14.8.0": {
+          "release_date": "2020-08-11",
+          "release_notes": "https://nodejs.org/en/blog/release/v14.8.0/",
+          "status": "retired",
```
Comment (vinyldarkscratch) [inline browsers/nodejs.json:275]:
```suggestion
          "status": "esr",
```

**Review by vinyldarkscratch — CHANGES_REQUESTED** [review]
Thanks for cherry-picking these changes!  Since NodeJS v14 is still in Active LTS, can we mark the most recent v14 version as "esr"?

(Note: I updated the title of this PR since it includes fixes for other NodeJS version statuses.)

**Comment by saschanaz** [comment]
14.8 is not most recent, 14.15 is. But then we probably don't want to add every version that we'll never use? 🤔

File: browsers/nodejs.json (hunk 1 above)
Comment (saschanaz) [reply browsers/nodejs.json:275]: 14.8 is not most recent, 14.15 is. But then we probably don't want to add every version that we'll never use? What do you think?

Comment (vinyldarkscratch) [reply browsers/nodejs.json:275]: Yeah, I've generally stuck with the lowest versions for each column on https://node.green/ with some exceptions (such as major versions or support differences undocumented on node.green).

**Review by vinyldarkscratch — APPROVED** [review]
Thanks, looking good to me!

**Pull request merged**